services/ui_backend_service/data/refiner/artifact_refiner.py

```diff
@@ -28,48 +18,43 @@ def __init__(self, cache):
         super().__init__(field_names=["content"], cache=cache)
 
     async def refine_record(self, record):
-        locations = [record[field] for field in self.field_names if field in record]
-        fetch_error = None
-        try:
-            data = await self.fetch_data(locations)
-        except GetArtifactsFailed as ex:
-            data = {}
-            fetch_error = format_error_body(getattr(ex, "id", None), str(ex))
-
-        _rec = record
-        for k, v in _rec.items():
-            if k in self.field_names:
-                _rec[k] = data[v] if v in data else None
-        if fetch_error:
-            _rec["postprocess_error"] = fetch_error
-        return _rec
+        _recs = self.refine_records([record])
+        return _recs[0] if len(_recs) > 0 else record
 
     async def refine_records(self, records):
         locations = [record[field] for field in self.field_names for record in records if field in record]
-        fetch_error = None
-        try:
-            data = await self.fetch_data(locations)
-        except GetArtifactsFailed as ex:
-            data = {}
-            fetch_error = format_error_body(getattr(ex, "id", None), str(ex))
+        errors = {}
+
+        def _event_stream(event):
+            if event.get("type") == "error" and event.get("key"):
+                loc = artifact_location_from_key(event["key"])
+                errors[loc] = event
+
+        responses = await self.fetch_data(locations, event_stream=_event_stream)
```

Comment:
is there currently any event stream that can come out of the cache action? or is this a remnant from some testing of streaming errors?

Reply:
As of now, there's no error streaming enabled for any of the actions.

I recommend leaving this as is, since the streaming errors are very useful for future, let's say we fetch all the timeline tasks via websocket request, we could essentially broadcast errors in realtime instead of reporting them after each batch is completed.